frontend/deb/debroot.go

```diff
@@ -0,0 +1,226 @@
+package deb
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"path/filepath"
+	"runtime/debug"
+
+	"github.com/Azure/dalec"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/pkg/errors"
+)
+
+//go:embed templates/patch-header.txt
+var patchHeader []byte
+
+func sourcePatchesDir(sOpt dalec.SourceOpts, base llb.State, dir, name string, spec *dalec.Spec, opts ...llb.ConstraintsOpt) ([]llb.State, error) {
```

Comment:
maybe a comment here explaining what this function does?

Reply:
Done.